In [3]:
!which python

/Users/brayan/jupyter-env/bin/python


In [6]:
!pip install pandas numpy matplotlib seaborn

   ━━━━━━╺━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.8/11.6 MB 2.6 MB/s eta 0:00:04
Resuming download pandas-2.3.3-cp39-cp39-macosx_10_9_x86_64.whl (1.8 MB/11.6 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.6/11.6 MB 220.8 kB/s  0:00:37 eta 0:00:02
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 21.2/21.2 MB 380.8 kB/s  0:00:51m0:00:0100:02
   ━━╸━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.5/7.9 MB 203.7 kB/s eta 0:00:37
Resuming download matplotlib-3.9.4-cp39-cp39-macosx_10_12_x86_64.whl (524 kB/7.9 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.9/7.9 MB 237.6 kB/s  0:00:30 eta 0:00:02
   ━━━━╺━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.3/2.3 MB ? eta -:--:--
Resuming download fonttools-4.60.1-cp39-cp39-macosx_10_9_x86_64.whl (262 kB/2.3 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.3/2.3 MB 154.2 kB/s  0:00:13 eta 0:00:02
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.3/5.3 MB 297.0 kB/s  0:00:17m0:00:0100:02
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 13/13 [seaborn]2/13 [seaborn]ib]


In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sb
import numpy as np

In [5]:
!pip install openpyxl

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2/2 [openpyxl]1/2 [openpyxl]


In [6]:
diccionario = pd.read_excel('diccionario_formato_911.xlsx')

In [9]:
diccionario.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 156 entries, 0 to 155
Data columns (total 2 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   NOMBRE       156 non-null    object
 1   DESCRIPCION  156 non-null    object
dtypes: object(2)
memory usage: 2.6+ KB


In [10]:
diccionario.head(10)

NOMBRE                      DESCRIPCION
0      ENTIDAD              CLAVE DE LA ENTIDAD
1    N_ENTIDAD             NOMBRE DE LA ENTIDAD
2    MUNICIPIO              CLAVE DEL MUNICIPIO
3   N_MUNICIPI             NOMBRE DEL MUNICIPIO
4    LOCALIDAD            CLAVE DE LA LOCALIDAD
5  N_LOCALIDAD           NOMBRE DE LA LOCALIDAD
6    DOMICILIO  DIRECCION DEL CENTRO DE TRABAJO
7     CLAVECCT                    CLAVE DEL CCT
8        N_CCT                   NOMBRE DEL CCT
9        TURNO                  CLAVE DEL TURNO

In [11]:
datos_2019_2020 = pd.read_csv('formato_911_basica_2019-2020.csv')

In [ ]:
import pandas as pd

# Specify the path to your Excel file
excel_file_path = 'your_excel_file.xlsx'

# Create an ExcelFile object
xls = pd.ExcelFile(excel_file_path)

# Access the sheet_names attribute to get a list of all sheet names
sheet_names_list = xls.sheet_names

# Print the list of sheet names
print(sheet_names_list)

In [12]:
datos_2019_2020.head()

entidad       n_entidad  municipio                 n_municipi  localidad  \
0        1  AGUASCALIENTES          3                   CALVILLO         33   
1        1  AGUASCALIENTES          1             AGUASCALIENTES       1097   
2        1  AGUASCALIENTES          6        PABELLÓN DE ARTEAGA         10   
3        1  AGUASCALIENTES          1             AGUASCALIENTES       1025   
4        1  AGUASCALIENTES         11  SAN FRANCISCO DE LOS ROMO         84   

                             n_localidad                  domicilio  \
0                     JALTICHE DE ARRIBA      CALLE RINCON DE ROMOS   
1  HACIENDA SAN MARCOS [FRACCIONAMIENTO]      AVENIDA DE LAS PRESAS   
2                            EL GARABATO  CALLE VENUSTIANO CARRANZA   
3                                POCITOS        AVENIDA JESUS MARIA   
4                           EL CHAPARRAL            NINGUNO NINGUNO   

     clavecct                            n_cct  turno  ... doc_1 doc_2 doc_3  \
0  01DPR0270P        SOR JUANA INES DE LA CRUZ      1  ...     0     0     1   
1  01DPR0733G  RAFAEL FRANCISCO AGUILAR LOMELÍ      1  ...     2     1     1   
2  01DPR0175L                 SALVADOR LUEVANO      1  ...     0     0     0   
3  01DTV0067C      ETV NUM. 67 MIGUEL BARRAGAN      1  ...     3     3     2   
4  01KJN0308E           PREESCOLAR COMUNITARIO      1  ...     0     0     0   

  doc_4 doc_5 doc_6 doc_mg doc_tot  tipo_org_docente         tipo_org_alumnos  
0     0     0     1      2       4        MULTIGRADO               INCOMPLETA  
1     1     1     1      0       7    COMPLEMENTARIA                 COMPLETA  
2     0     0     0      2       2        MULTIGRADO               INCOMPLETA  
3     0     0     0      0       8    COMPLEMENTARIA  NO APLICA EN ESTE NIVEL  
4     0     0     0      1       1        MULTIGRADO               INCOMPLETA  

[5 rows x 156 columns]

In [14]:
for i in range(19,25,1):
    print(f'formato_911_basica_20{i}-20{i+1}.csv')
#datos_2019_2020 = pd.read_csv(f'formato_911_basica_20{i}-20{i+1}.csv')

formato_911_basica_2019-2020.csv
formato_911_basica_2020-2021.csv
formato_911_basica_2021-2022.csv
formato_911_basica_2022-2023.csv
formato_911_basica_2023-2024.csv
formato_911_basica_2024-2025.csv


In [18]:
import pandas as pd
import glob
import os

In [19]:
csv_files = sorted(glob.glob("formato_911_basica_*.csv"))
csv_files

['formato_911_basica_2019-2020.csv',
 'formato_911_basica_2020-2021.csv',
 'formato_911_basica_2021-2022.csv',
 'formato_911_basica_2022-2023.csv',
 'formato_911_basica_2023-2024.csv',
 'formato_911_basica_2024-2025.csv']

In [20]:
dataframes = {}

for file in csv_files:
    # Extraer el nombre del archivo sin extensión
    year = os.path.splitext(os.path.basename(file))[0].replace("formato_911_basica_", "")
    
    # Leer el CSV
    df = pd.read_csv(file)
    
    # Guardar en el diccionario
    dataframes[year] = df

# Ejemplo de acceso
dataframes["2022-2023"].head()

entidad       n_entidad  municipio                 n_municipi  localidad  \
0        1  AGUASCALIENTES          1             AGUASCALIENTES          1   
1        1  AGUASCALIENTES          1             AGUASCALIENTES          1   
2        1  AGUASCALIENTES          2                   ASIENTOS         19   
3        1  AGUASCALIENTES         11  SAN FRANCISCO DE LOS ROMO          1   
4        1  AGUASCALIENTES          1             AGUASCALIENTES          1   

                 n_localidad               domicilio    clavecct  \
0             AGUASCALIENTES  CALLE GENERAL BARRAGAN  01PJN0097M   
1             AGUASCALIENTES       AVENIDA DEL VALLE  01PPR0161N   
2               LA ESPERANZA        CALLE CENZONTLES  01KJN0089I   
3  SAN FRANCISCO DE LOS ROMO   CALLE 10 DE NOVIEMBRE  01DPR0017W   
4             AGUASCALIENTES         NINGUNO NINGUNO  01EPR0001U   

                                               n_cct  turno  ... doc_1 doc_2  \
0                             COLEGIO MÉXICO MODERNO      1  ...     1     1   
1                                     INSTITUTO CIMA      1  ...     1     1   
2                             PREESCOLAR COMUNITARIO      1  ...     0     0   
3                          J. REFUGIO MIRANDA AGUAYO      1  ...     2     2   
4  CENTRO EDUCATIVO INTEGRAL PARA ALTAS CAPACIDAD...      1  ...     0     0   

  doc_3 doc_4 doc_5 doc_6 doc_mg doc_tot  tipo_org_docente  tipo_org_alumnos  
0     1     0     0     0      0       3    COMPLEMENTARIA          COMPLETA  
1     1     1     1     1      0       6    COMPLEMENTARIA          COMPLETA  
2     0     0     0     0      1       1        MULTIGRADO        INCOMPLETA  
3     2     2     3     2      0      13    COMPLEMENTARIA          COMPLETA  
4     1     1     1     1      0       4    COMPLEMENTARIA        INCOMPLETA  

[5 rows x 156 columns]

In [21]:
for year, df in dataframes.items():
    print(year, df.shape)

2019-2020 (230424, 156)
2020-2021 (228852, 156)
2021-2022 (228804, 156)
2022-2023 (229691, 156)
2023-2024 (231534, 156)
2024-2025 (231913, 156)


Todos los dataframes tienen 156 columnas, aunque difieren en el número de filas.

In [24]:
# Combinar todos los DataFrames y añadir la columna 'periodo_escolar'
combined_df = pd.concat(
    [df.assign(periodo_escolar=year) for year, df in dataframes.items()],
    ignore_index=True
)

In [25]:
combined_df.info()
combined_df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1381218 entries, 0 to 1381217
Columns: 313 entries, entidad to TIPO_ORG_ALUMNOS
dtypes: float64(280), object(33)
memory usage: 3.2+ GB


entidad       n_entidad  municipio                 n_municipi  localidad  \
0      1.0  AGUASCALIENTES        3.0                   CALVILLO       33.0   
1      1.0  AGUASCALIENTES        1.0             AGUASCALIENTES     1097.0   
2      1.0  AGUASCALIENTES        6.0        PABELLÓN DE ARTEAGA       10.0   
3      1.0  AGUASCALIENTES        1.0             AGUASCALIENTES     1025.0   
4      1.0  AGUASCALIENTES       11.0  SAN FRANCISCO DE LOS ROMO       84.0   

                             n_localidad                  domicilio  \
0                     JALTICHE DE ARRIBA      CALLE RINCON DE ROMOS   
1  HACIENDA SAN MARCOS [FRACCIONAMIENTO]      AVENIDA DE LAS PRESAS   
2                            EL GARABATO  CALLE VENUSTIANO CARRANZA   
3                                POCITOS        AVENIDA JESUS MARIA   
4                           EL CHAPARRAL            NINGUNO NINGUNO   

     clavecct                            n_cct  turno  ... DOC_1 DOC_2 DOC_3  \
0  01DPR0270P        SOR JUANA INES DE LA CRUZ    1.0  ...   NaN   NaN   NaN   
1  01DPR0733G  RAFAEL FRANCISCO AGUILAR LOMELÍ    1.0  ...   NaN   NaN   NaN   
2  01DPR0175L                 SALVADOR LUEVANO    1.0  ...   NaN   NaN   NaN   
3  01DTV0067C      ETV NUM. 67 MIGUEL BARRAGAN    1.0  ...   NaN   NaN   NaN   
4  01KJN0308E           PREESCOLAR COMUNITARIO    1.0  ...   NaN   NaN   NaN   

  DOC_4 DOC_5 DOC_6 DOC_MG DOC_TOT  TIPO_ORG_DOCENTE  TIPO_ORG_ALUMNOS  
0   NaN   NaN   NaN    NaN     NaN               NaN               NaN  
1   NaN   NaN   NaN    NaN     NaN               NaN               NaN  
2   NaN   NaN   NaN    NaN     NaN               NaN               NaN  
3   NaN   NaN   NaN    NaN     NaN               NaN               NaN  
4   NaN   NaN   NaN    NaN     NaN               NaN               NaN  

[5 rows x 313 columns]

In [27]:
#Verificando los valores de la columna "periodo_escolar"
combined_df['periodo_escolar'].unique()

array(['2019-2020', '2020-2021', '2021-2022', '2022-2023', '2023-2024',
       '2024-2025'], dtype=object)

Ahora los 6 archivos .csv están cargados todos en un solo DataFrame, combined_df, y existe una nueva columna identificadora llamada "periodo_escolar".